go/tools/gazelle/rules/generator.go

```diff
@@ -105,11 +105,13 @@ func (g *generator) Generate(rel string, pkg *build.Package) ([]*bzl.Rule, error
 	}
 
 	library := defaultLibName
-	r, err := g.generateLib(rel, library, pkg, cgoLibrary)
+	libRule, err := g.generateLib(rel, library, pkg, cgoLibrary)
 	if err != nil {
 		return nil, err
 	}
-	rules = append(rules, r)
+	if libRule != nil {
+		rules = append(rules, libRule)
```

Comment:
Does the `go_binary` rule depend on an undefined library rule if `libRule` is `nil`?
IIUC, it can happen when the command package consists of only cgo-enabled sources.